File: shoc-containerize/src/main/java/io/imast/shoc/containerize/RegistryDefinition.java
package io.imast.shoc.containerize;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * The registry reference
 * 
 * @author Davit.Petrosyan
 */
@Data
@Builder(toBuilder = true)
@AllArgsConstructor
@NoArgsConstructor
public class RegistryDefinition {
    
    /**
     * The docker registry
     */
    private String registry;
    
    /**
     * The repository to group images
     */
    private String repository;
    
    /**
     * The docker registry username
     */
    private String username;
    
    /**
     * The docker registry password
     */
    private String password;
    
    /**
     * The docker registry email
     */
    private String email;
}

File: Shoc.Engine.Model/RegistryDefinition.cs
namespace Shoc.Engine.Model
{
    /// <summary>
    /// The registry definition
    /// </summary>
    public class RegistryDefinition
    {
        /// <summary>
        /// The registry identity
        /// </summary>
        public string Id { get; set; }

        /// <summary>
        /// The registry Uri
        /// </summary>
        public string RegistryUri { get; set; }

        /// <summary>
        /// The repository
        /// </summary>
        public string Repository { get; set; }

        /// <summary>
        /// The registry email
        /// </summary>
        public string Email { get; set; }

        /// <summary>
        /// The registry username
        /// </summary>
        public string Username { get; set; }

        /// <summary>
        /// The registry password
        /// </summary>
        public string Password { get; set; }
    }
}